### PR TITLE
Added an endpoint to create a valid Firebase user auth token

### PR DESCRIPTION
**Pull request by gabek**
This is my first Python thing.  Tabs are hard.

It feels weird to put an API endpoint in a file called *"views.py"*, but I did it anyway because I don't know where else to put it.
The Django documentation states that I can specify a *status_code* in the *HttpResponse* but it doesn't seem to work?  It'd be nice to throw a legit HTTP 400 on an error.

Thank you for your consideration.  I hope this pull request finds you well.

### CODE REVIEW

File: rdiodj/views.py [hunk 1]
```diff
@@ -226,6 +227,22 @@ def ajax(request):
 
     return HttpResponse(json.dumps(data), content_type = "application/json")
 
+def createauthtoken(request):
+    response = None
+
+    rdio_user_key = request.GET.get('userKey')
+    if rdio_user_key is None:
```
Comment (jackdbernier) [inline rdiodj/views.py]: In python None is interpreted as falsely.

```
    if rdio_user_key:
        custom_data = {'rdio_user_key': rdio_user_key}
        options = {'debug': settings.DEBUG}
        firebase_token = create_token(settings.FIREBASE_TOKEN, custom_data, options)
        response = { "token": firebase_token }
    else:
        response = {"error": "userKey is a required GET param"}
```

**Comment by jackdbernier** [comment]
We are using a old deprecated version of django... maybe that's why.

**Comment by dasevilla** [comment]
I would feel better if you just implemented Firebase directly. An API method that hands out tokens without auth doesn't sound good.

**Comment by gabek** [comment]
That was my original plan, but Firebase actually said in it's documentation to go this route:

```Firebase JWTs should always be generated on a trusted server so that the Firebase Secret which is needed to generate them can be kept private.```

Given that Firebase custom auth libraries were created for server-side environments and not client-side solidified it.  I could build my own client-side one of course, but only if that's best practice.

**Comment by mkapolka** [comment]
This is basically the same thing we're doing in the front end already. Eventually we'll want to do a proper API without handing out firebase keys, but that will happen when we drop firebase entirely. This looks good to me.

**Comment by dasevilla** [comment]
:shipit:

**Comment by gabek** [comment]
Thanks, all!

**Comment by rstecker** [comment]
err-- given the shipit squirrel I assume this was meant to be merged...?

**Pull request merged**